File: app/routers/session.py
from fastapi import APIRouter, Depends

from app.auth import require_identity
from app.models import Identity, MeResponse


router = APIRouter(prefix="/api", tags=["Session"])


@router.get("/me", response_model=MeResponse)
async def me(identity: Identity = Depends(require_identity)) -> MeResponse:
    return MeResponse(id=identity.id, role=identity.role)

File: server/routes/session.js
import express from "express";

import { requireIdentity } from "../auth.js";

export function createSessionRouter(store) {
  const router = express.Router();

  router.get("/me", requireIdentity(store), (req, res) => {
    res.json(req.identity);
  });

  return router;
}
